pennylane/ops/qubit/matrix_ops.py

```diff
@@ -270,9 +271,8 @@ def _eigvals(cls, *params):
 
         return D
 
-    @staticmethod
-    def decomposition(D, wires):
-        return [QubitUnitary(qml.math.diag(D), wires=wires)]
+    def decomposition(self):
+        return [QubitUnitary(qml.math.diag(self.parameters[0]), wires=self.wires)]
```

Comment:
I know, this pattern that the developer needs to know that `parameters[0]` is `D` is not optimal, but I can't think of a better option.

Reply:
On the other hand, if PennyLane feeds `op.parameters` into `decomposition` then PennyLane makes the assumption that `D` is `parameters[0]`, which is probably a lot less safe!